queue.go

```diff
@@ -94,18 +95,16 @@ type Queue struct {
 }
 
 // NewQueue creates and returns a new Queue structure
-func NewQueue(url string, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
+func NewQueue(name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
 	q := &Queue{
-		name: name,
-		url:  url,
-
+		name:         name,
 		isConsumer:   isConsumer,
 		Durable:      durable,
 		Jobs:         jobs,
 		prefetchSize: prefetchSize,
 		workers:      make([]worker, 0),
 	}
-
+	q.setUrl()
```

Comment:
instead of calling method, you can initialize ( and check for empty ) all the variables once in the `func init() { ... }` or a separate config file can be used.

so instead of `q.setUrl()` you can use  `fmt.Sprintf(urlString, userName, password, hostName, port, vHost)` in the `q` initialization itself.

it will make sure NewQueue is using same values for those variables ( unless having different values is not the requirement ).

Reply:
what will happen, if someone need to use 2 queues in one app? Calling `init()` and panicking is not a reusable nor pretty choice for modules, which can be used by different applications